### PR TITLE
Better optimistic response for inline reacts

### PR DESCRIPTION
**Pull request by Raemon**
The fact that inline reacts took a few seconds to appear at the bottom of the comment was pretty annoying and confusing UI. This PR moves the voteProps higher up in the component-tree so that the VotingBottom component and the Inline reacts component can share optimistic response.

Some potential tricksiness to think about:
– I move voteProps up to the commentItem component. In theory, it could passed down into the SmallVote item now. But, SmallVote was previously designed to be self-sufficient (i.e. has it's own useVote hook), and I think probably we should just leave it as-is
– VoteBottomComponent now needs to take in voteProps, and VoteBottomComponent currently shares a type interface with the other vote components, so it's a bit awkward that I currently have voteProps as optional on CommentVotingComponentProps but required in the extended interface of NamesAttachedReactionsCommentBottomProps. 

In the process, the VotingProps type ended up in a circular dependency. I've currently pulled that out to another file in the same directory. It's possible it should live somewhere else but I wasn't sure where else to put it.

The current calls I made here were my best guess but I'm not very confident in them.

### CODE REVIEW

**Comment by b0b3rt** [comment]
> – VoteBottomComponent now needs to take in voteProps, and VoteBottomComponent currently shares a type interface with the other vote components, so it's a bit awkward that I currently have voteProps as optional on CommentVotingComponentProps but required in the extended interface of NamesAttachedReactionsCommentBottomProps.

Yeah, the fact that Typescript lets you assign `() => Components.NamesAttachedReactionsCommentBottom` to `getCommentBottomComponent`, which has a type of `() => CommentVotingComponent`, and `NamesAttachedReactionsCommentBottom` requires `voteProps` while they're optional in `CommentVotingComponent`, is a result of Typescript's unsoundness around optional function parameters.  If we set `"strictFunctionTypes": true` in our `tsconfig` it'd catch it, but... we'd have to fix a lot of type issues.

Still, in this case, it's actually a footgun; I'd probably avoid typing `NamesAttachedReactionsCommentBottom` with required `voteProps`.  Better to either leave them optional and figure out a fall-back in the component if they aren't passed in, or make them required for all `CommentVotingComponent`s and then just ignore them in those where you don't need them.

**Comment by Raemon** [comment]
Hmm. I do immediately need the voteProps for a hook, and because it's a hook, I can't wait until after it to do a check for voteProps or return null.

```interface NamesAttachedReactionsCommentBottomProps extends CommentVotingComponentProps {
  voteProps: VotingProps<VoteableTypeClient>,
}

const NamesAttachedReactionsCommentBottom = ({
  document, hideKarma=false, commentItemRef, classes, voteProps
}: CommentVotingComponentProps & WithStylesProps) => {
  const anchorEl = useRef<HTMLElement|null>(null);
  const currentUser = useCurrentUser();

  const { getAlreadyUsedReactTypesByKarma, getAllReactionQuotes } = useNamesAttachedReactionsVoting(voteProps)

  const extendedScore = voteProps.document?.extendedScore as NamesAttachedReactionsScore|undefined;
  const allReactions = getAlreadyUsedReactTypesByKarma();

  const visibleReactionsDisplay = reactionsListToDisplayedNumbers(extendedScore?.reacts ?? null, currentUser?._id);
  ```

**Comment by Raemon** [comment]
Okay, I maybe fixed this via splitting CommentVotingComponent into a separate CommentVotingBottomComponent with different types, and requiring voteProps there.

File: packages/lesswrong/components/votes/votingProps.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/votingProps.ts:1]: Extra line

File: packages/lesswrong/components/votes/lwReactions/ReactionQuotesHoverInfo.tsx [hunk 2]
```diff
@@ -3,10 +3,10 @@ import { NamesAttachedReactionsList, NamesAttachedReactionsScore } from '../../.
 import { useCurrentUser } from '../../common/withUser';
 import { registerComponent, Components } from '../../../lib/vulcan-lib';
 import { clearHighlights, markHighlights, useNamesAttachedReactionsVoting } from './NamesAttachedReactionsVoteOnComment';
-import { VotingProps } from '../withVote';
 import filter from 'lodash/filter';
 import uniq from 'lodash/uniq';
 import { dimHighlightClassName, highlightSelectorClassName } from '../../comments/CommentsItem/CommentsItem';
+import { VotingProps } from '../votingProps';
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/lwReactions/ReactionQuotesHoverInfo.tsx:9]: nit: undo unnecessary diff

File: packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React, { useState, useRef, RefObject, useEffect } from 'react';
 import { Components, registerComponent } from '../../../lib/vulcan-lib';
-import { CommentVotingComponentProps, } from '../../../lib/voting/votingSystems';
+import { CommentVotingComponentProps, NamesAttachedReactionsCommentBottomProps, } from '../../../lib/voting/votingSystems';
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx:3]: still using `CommentVotingComponentProps`?

File: packages/lesswrong/components/votes/lwReactions/InlineReactSelectionWrapper.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import Mark from 'mark.js';
 import React, { useEffect, useState, useRef, useCallback } from 'react';
 import { getVotingSystemByName } from '../../../lib/voting/votingSystems';
 import { registerComponent, Components } from '../../../lib/vulcan-lib';
+import { VotingProps } from '../votingProps';
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/lwReactions/InlineReactSelectionWrapper.tsx:5]: can be `import type`

File: packages/lesswrong/components/votes/lwReactions/AddInlineReactionButton.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import React, { useRef, useState } from "react";
 import { Components, registerComponent } from "../../../lib/vulcan-lib";
-import { VotingProps } from "../withVote";
 import InsertEmoticonOutlined from '@material-ui/icons/InsertEmoticon';
 import { useNamesAttachedReactionsVoting } from "./NamesAttachedReactionsVoteOnComment";
+import { VotingProps } from "../votingProps";
```
Comment (b0b3rt) [inline packages/lesswrong/components/votes/lwReactions/AddInlineReactionButton.tsx:5]: nit: undo unnecessary diff

Comment (b0b3rt) [reply packages/lesswrong/components/votes/lwReactions/AddInlineReactionButton.tsx:5]: oops, nvm, it's in a new file.

File: packages/lesswrong/components/votes/lwReactions/ReactionQuotesHoverInfo.tsx (hunk 2 above)
Comment (b0b3rt) [reply packages/lesswrong/components/votes/lwReactions/ReactionQuotesHoverInfo.tsx:9]: nvm, it's in a new file.

File: packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx (hunk 3 above)
Comment (Raemon) [reply packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx:3]: Yes, for the top voting component
